pallets/token-timeout/src/lib.rs

```diff
@@ -0,0 +1,327 @@
+		StorageMap<_, Blake2_256, T::AccountId, AccountTimeoutDataInfo<T::BlockNumber>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		TimeoutMetadataUpdated,
+		TimeoutReleased(T::AccountId, Balance),
+	}
+
+	#[pallet::error]
+	/// Errors
+	pub enum Error<T> {
+		/// Timeouts were incorrectly initialized
+		TimeoutsIncorrectlyInitialzed,
+		/// Timeout metadata is invalid
+		InvalidTimeoutMetadata,
+		/// Timeouts have not been initialzed
+		TimeoutsNotInitialized,
+		/// No tokens of the user are timedout
+		NotTimedout,
+		/// The timeout cannot be released yet
+		CantReleaseYet,
+		/// The limit on the maximum curated tokens for which there is a swap threshold is exceeded
+		MaxCuratedTokensLimitExceeded,
+		/// An unexpected failure has occured
+		UnexpectedFailure,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		#[pallet::constant]
+		type MaxCuratedTokens: Get<u32>;
+		type Tokens: MultiTokenCurrencyExtended<Self::AccountId>
+			+ MultiTokenReservableCurrency<Self::AccountId>;
+		#[pallet::constant]
+		type NativeTokenId: Get<TokenId>;
+		type WeightInfo;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[transactional]
+		#[pallet::weight(1_000_000_000)]
+		pub fn update_timeout_metadata(
+			origin: OriginFor<T>,
+			period_length: Option<T::BlockNumber>,
+			timeout_amount: Option<Balance>,
+			swap_value_thresholds: Option<Vec<(TokenId, Option<Balance>)>>,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			let mut timeout_metadata =
+				Self::get_timeout_metadata().unwrap_or(TimeoutMetadataInfo {
+					period_length: Default::default(),
+					timeout_amount: Default::default(),
+					swap_value_threshold: BoundedBTreeMap::new(),
+				});
+
+			timeout_metadata.period_length =
+				period_length.unwrap_or(timeout_metadata.period_length);
+			timeout_metadata.timeout_amount =
+				timeout_amount.unwrap_or(timeout_metadata.timeout_amount);
+
+			ensure!(!timeout_metadata.period_length.is_zero(), Error::<T>::InvalidTimeoutMetadata);
+			ensure!(!timeout_metadata.period_length.is_zero(), Error::<T>::InvalidTimeoutMetadata);
+
+			if let Some(swap_value_thresholds) = swap_value_thresholds {
+				for (token_id, maybe_threshold) in swap_value_thresholds.iter() {
+					match maybe_threshold {
+						Some(threshold) => {
+							let _ = timeout_metadata
+								.swap_value_threshold
+								.try_insert(*token_id, *threshold)
+								.map_err(|_| Error::<T>::MaxCuratedTokensLimitExceeded)?;
+						},
+						None => {
+							let _ = timeout_metadata.swap_value_threshold.remove(token_id);
+						},
+					}
+				}
+			}
+
+			TimeoutMetadata::<T>::put(timeout_metadata);
+
+			Pallet::<T>::deposit_event(Event::TimeoutMetadataUpdated);
+
+			Ok(().into())
+		}
+
+		#[transactional]
+		#[pallet::weight(1_000_000_000)]
```

Comment:
what about benchmarks ? will they come in separate pr ?

Reply:
they're here